sync/src/api.rs

```diff
@@ -496,6 +496,8 @@ pub trait ManageNetwork : Send + Sync {
 	fn stop_network(&self);
 	/// Query the current configuration of the network
 	fn network_config(&self) -> NetworkConfiguration;
+	/// Get network context for protocol.
+	fn with_proto_context(&self, proto: ProtocolId, f: &mut FnMut(&NetworkContext));
```

Comment:
can you think of better way of extending this api which could be interprocess one day?

Reply:
The main problem it solves is creating a pathway for the whisper RPC to post messages to the whisper network. Extending the API to make that possible would require us to come up with a cross-binary network context, which is a desirable feature in general. Changing the signature to `fn proto_context(&self, proto: ProtocolId) -> Option<&NetworkContext>` could work, but would require pretty large changes to the inner workings.